app/src/main/java/com/android/voyageur/ui/trip/Tabs.kt

```diff
@@ -77,30 +78,49 @@ fun TopTabs(
         modifier = Modifier.fillMaxWidth().testTag("tabRow"),
     ) {
       // Create each tab with a Tab composable
-      tabs.forEachIndexed { index, title ->
-        Tab(
-            selected = navigationActions.getNavigationState().currentTabIndexForTrip == index,
-            onClick = { navigationActions.getNavigationState().currentTabIndexForTrip = index },
-            text = { Text(title) })
+      if (navigationActions.getNavigationState().isReadOnlyView) {
+        readOnlyTabs.forEachIndexed { index, title ->
+          Tab(
+              selected = navigationActions.getNavigationState().currentTabIndexForTrip == index,
+              onClick = { navigationActions.getNavigationState().currentTabIndexForTrip = index },
+              text = { Text(title) })
+        }
+      } else {
+        tabs.forEachIndexed { index, title ->
+          Tab(
+              selected = navigationActions.getNavigationState().currentTabIndexForTrip == index,
+              onClick = { navigationActions.getNavigationState().currentTabIndexForTrip = index },
+              text = { Text(title) })
+        }
       }
     }
 
-    // Display content based on selected tab
-    when (navigationActions.getNavigationState().currentTabIndexForTrip) {
-      0 -> ScheduleScreen(tripsViewModel, selectedTrip, navigationActions, userViewModel)
-      1 -> ActivitiesScreen(navigationActions, userViewModel, tripsViewModel)
-      2 -> PhotosScreen(tripsViewModel, navigationActions, userViewModel)
-      3 ->
-          SettingsScreen(
-              selectedTrip,
-              navigationActions,
-              tripsViewModel = tripsViewModel,
-              userViewModel = userViewModel,
-              placesViewModel = placesViewModel,
-              onUpdate = {
-                navigationActions.getNavigationState().currentTabIndexForTrip = 0
-                navigationActions.getNavigationState().currentTabIndexForTrip = 3
-              })
+    when (navigationActions.getNavigationState().isReadOnlyView) {
+      true -> {
+        when (navigationActions.getNavigationState().currentTabIndexForTrip) {
+          // Pass true for the isReadOnly parameters
+          0 -> ScheduleScreen(tripsViewModel, selectedTrip, navigationActions, userViewModel, true)
+          1 -> ActivitiesScreen(navigationActions, userViewModel, tripsViewModel, true)
```

Comment:
It's not really clear to me why we pass isReadOnlyView as a parameter when we already pass navigationActions which has the readOnlyView property?

Reply:
I see what you mean. My thought was that by not passing it as a parameter we can avoid situations where `navigationActions.getNavigationState().isReadOnlyView` is true, but the parameter is false (or the other way around).
However, I understand how this would make testing more difficult. 
I'll approve this PR for now as this choice doesn't impact the app, and I'll leave it up to you. If you'd like, we can look more into this next sprint, I think I worked on some similar tests for the current tab field of the `navigtionActions` :)